### ISSUE TITLE
nsxt_distributed_firewall logging is always FALSE

### ISSUE THREAD

**Issue by storbel** [vmware/terraform-provider-vcd#985]
Hello,

### Terraform Version
Terraform v1.2.1

### Affected Resource(s)
Please list the resources as a list, for example:
- nsxt_distributed_firewall 

### Terraform Configuration Files

```hcl
rule {
    name        = "MSEG-CL-OP_F5-VIP-LAN-DDD-PROD"
    action      = "ALLOW"
    ip_protocol = "IPV4"
    direction   = "IN_OUT"
    enabled     = true
    logging     = true  <---------
    comment     = "-"

    app_port_profile_ids = [xxx ]
    destination_ids = [xxx, ]
    source_ids = [xxx ]

  }

```

### Debug Output

the **plan** command  detect the logging value changes and tells that it will update it, unfortunately after the **apply** command, nothing changes and the states show the logging false
the rule definition is well done with the logging = true
```shell
    rule {
        action                      = "ALLOW"
        app_port_profile_ids        = [
            "urn:vcloud:applicationPortProfile:7714d75c-31bc-47dd-870b-4f2d01847ec8",
        ]
        comment                     = "-"
        destination_groups_excluded = false
        destination_ids             = [
            "urn:vcloud:firewallGroup:64cd119b-a8c2-4f93-bebe-c0480b8a9eb1",
        ]
        direction                   = "IN_OUT"
        enabled                     = true
        id                          = "aae4c87e-68a6-4275-ab24-bd5a0fb8a5b2"
        ip_protocol                 = "IPV4"
        logging                     = false <----
        name                        = "MSEG-CL-OP_F5-VIP-LAN-DDD-PROD"
        network_context_profile_ids = []
        source_groups_excluded      = false
        source_ids                  = [
            "urn:vcloud:firewallGroup:f10fbd7d-5923-438d-994c-1c9f4d267c68",
        ]
    }
   
```
after the terraform apply, it remains false on the states and on the vcloud director console

### Expected Behavior
Create the rule with logging active

### Actual Behavior
The logging is disabled on the states and on the vcloud director console

**Comment by Didainius** [comment]
Hello @storbel ,
I have just tried to run this as a system administrator and it did get enabled. It also detected drifts after a `refresh` operation.
I remember from NSX-V firewalling, that to enable logging one had to have some specific right. If that right was not present - the API call with `true` for logging would simply get ignored and stay `false` (just like in your case).

One immediate thought I have without further investigation is - can you check that your role has this right `vDC Group: Configure Logging`? I think it is not added to Org admins by default. If this doesn't work out for you - I could dig a bit deeper to find this.

**Comment by Didainius** [comment]
The obvious thing I forgot to mention is - can you just check that you can enable rule logging in UI using that same user?

**Comment by storbel** [comment]
Hi @Didainius 
Thanks for your swift reply. 
I'm able to update the logging value from the console

**Comment by Didainius** [comment]
This is very odd if this was the same user.

I have done more validations and my hypothesis turned out to be correct for me. It all worked with Org Admin user for me, but as soon as I removed `vDC Group: Configure Logging` right from global org admin role (and published it to my org) - your described behavior happens.

<img width="1475" alt="image" src="https://user-images.githubusercontent.com/15804230/213440463-955e5247-fb0b-4e0d-9da1-b79e28e94b29.png">


Which VCD version is it? Could enable logging (https://registry.terraform.io/providers/vmware/vcd/latest/docs#logging) and then look for the payload that is being sent to VCD.

You should search for similar payload there (it will probably be easiest to look by rule name, if you make it unique enough):
```json
 {
      "id": "5dc9e13c-248a-4b56-a2a2-4e0d1505fde4",
      "name": "rule3",
      "description": null,
      "sourceFirewallGroups": null,
      "destinationFirewallGroups": null,
      "applicationPortProfiles": null,
      "ipProtocol": "IPV4",
      "action": "DROP",
      "actionValue": "DROP",
      "direction": "IN_OUT",
      "logging": true,
      "networkContextProfiles": null,
      "enabled": true,
      "version": {
        "version": 0
      },
      "comments": null,
      "sourceGroupsExcluded": false,
      "destinationGroupsExcluded": false
    },
```

**Comment by Didainius** [comment]
@storbel ,
I just reminded one more thing.
If you enable logging, you can see all the rights your user has while logging in. Look for something like:
```
2023/01/23 11:21:25 ********************************************************************************
2023/01/23 11:21:25 START SESSION INFO
2023/01/23 11:21:25 ********************************************************************************
2023/01/23 11:21:25 {
  "User": "administrator",
  "Org": "System",
  "Roles": [
   "System Administrator"
  ],
  "Rights": [
   "Group / User: Manage",
   "Organization vDC: View CPU and Memory Reservation",
   "Access Metrics Endpoint",
   "API Tokens: Manage",
   "API Tokens: Manage All",
   "vGPU Profile: Manage",
....
```

Check if the `vDC Group: Configure Logging` is in there

**Comment by Didainius** [comment]
@storbel ,
I am closing this issue - feel free to reopen if you still see something odd.

**Comment by melmou** [comment]
Hello, i have the same problem, i believe that this issue must be open again.

And yes @Didainius I have **vDC Group: Configure Logging** in my role, and i confirm that it's working from the UI, but through Terraform only disabling logs that works but enabling logs do not work unfortunately exactly like @storbel said.

Can anyone help ? 

**Note**: i have tested with 2 versions of the provider ( 3.8.2 and 3.10.0 ), and i have the same problem.

Thanks in advance

**Comment by Didainius** [comment]
@melmou,
That is odd. Could you attempt to check the log as we did above?

Also, could you attempt the "sister" `vcd_nsxt_distributed_firewall_rule` resource https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/nsxt_distributed_firewall_rule (it handles one rule per resource as opposed to all rules in one). I am curious if it has the same behavior

**Comment by melmou** [comment]
Yes indeed it's odd because only enabling that do not work with terraform, disabling works without problem.

Yes i checked the logs, and yes i can see **"vDC Group: Configure Logging"** in the rights.
And for the payload i can see that logging is true when sent to vcd.

Also i have tested [nsxt_distributed_firewall_rule] as you have suggested and unfortunately not only enabling still do not work, but also disabling logging do not work on this resource.

I should also note that i'm testing with inherited role from group ( with **"vDC Group: Configure Logging"** checked ofc ), but when i test using the role directly on the user without inheriting it from the group enabling logs works correctly without a problem.

**Comment by Didainius** [comment]
> I should also note that i'm testing with inherited role from group ( with **"vDC Group: Configure Logging"** checked ofc ), but when i test using the role directly on the user without inheriting it from the group enabling logs works correctly without a problem.

Is this exactly the same user that works in the UI? Which Cloud Director version are you using?

**Comment by melmou** [comment]
yes it's the same user that works in the UI
I'm using VCD 10.4

**Comment by Didainius** [comment]
> I'm using VCD 10.4

Is it 10.4.0?

**Comment by melmou** [comment]
It's **10.4.2.1** to be precise

**Comment by Jackpoto** [comment]
Hello as discussed with @melmou it seems to be something known on vCD side and resolved on last 10.4.2.2 release !

I just looked at the resolved issues of last release :
The VMware Cloud Director API query for user privileges on a set of entities and user granted privileges returns incorrect results

If an imported user is assigned the Defer to Identity Provider role and you run the VMware Cloud Director API to verify the user privileges on a set of entities or the user granted privileges, the API returns a reduced or an empty set of rights for this user.

We are going to test it asap.

Regards
